README.md

```diff
@@ -1179,6 +1186,44 @@ Prefer: return=minimal
 HTTP/1.1 204 No Content
 ```
 
+## HTTP Methods
+
+Following HTTP Methods are supported currently.
+
+### HEAD
+The same as [GET](#get), except that it doesn't return any body.
+
+### GET
+Used to query a resource with its sub/embedded resources.
+
+### POST
+Used to create new resource document, where new `ID` is generated from the server.
+
+### PUT
+Used to create/update single resource document given its `ID`. Be aware when dealing with resource fields with `Default` set. Initial creation for such resources will set particular field to its default value if omitted, however on subsequent `PUT` calls this field will be deleted if omitted. If persistent `Default` field is needed use `{Required: true}` with it. 
+
+### PATCH
+Used to update/patch single resource document given its `ID`. REST Layer supports following update protocols:
+
+- Simple filed replacement [RFC-5789](http://tools.ietf.org/html/rfc5789) - this protocol will udpate only supplied top level fields, and will leave other fields in the document intact. This means that this protocol can't delete fields. Using this protocol is specified with `Content-Type: application/json` HTTP Request header.
+
+- [JSON-Patch/RFC-6902](https://tools.ietf.org/html/rfc6902) - When patching deeply nested documents, it is more convenient to use protocol designed especially for this. Using this protocol is specified with `Content-Type: application/json-patch+json` HTTP Request header.
+
+	If using `If-Match` concurrency control as described in the [data control and integrity section](#data-integrity-and-concurrency-control), you could potentially choose to calculate the body of new object client side. Note that the response body for a successful operation can be omitted by supplying a HTTP request header: `Prefer: return=minimal`.
```

Comment:
Is this indent by a tab for `If using...` accidental?

Reply:
It is tabbed because his paragraph goes to the last JSON patch section.

Reply:
There is a preview function in the GitHub Code-Review (and in [VS Code](https://code.visualstudio.com/docs/languages/markdown)). It looks like it _works_, but I think it' looks a bit weird to have two paragraphs with newline spacing within a bullet-point. Would probably look better to just use paragraphs, or to some extent even tables.

I noticed just now, that there is another section called "Modes" that lists the HTTP Methods already. Should probably go over the README at some point and consolidate some sections...

For now, could you perhaps make the If-Match paragraph on independent of the bullet-point, and indent it to the left? I feel the text as it stands applies to both cases anyway; replacing a single-top level field is even easier to calculate client-side than to apply a JSON-Patch.

Same for the code block.

Reply:
Yes, you are actually right, it relates to both ways of patching. Done.
VSCode has build-in Markdown preview, and I am using it to verify my changes. I think it matches GHs Markdown processor entirely.